src/GitHub/Endpoints/Repos/Collaborators.hs

```diff
@@ -78,6 +78,6 @@ addCollaboratorR
     :: Name Owner        -- ^ Repository owner
     -> Name Repo         -- ^ Repository name
     -> Name User         -- ^ Collaborator to add
-    -> GenRequest 'MtUnit 'RW ()
+    -> GenRequest 'MtJSON 'RW RepoInvitation
```

Comment:
this will fail when https://developer.github.com/v3/repos/collaborators/#response-when-person-is-already-a-collaborator

is there value in returning invitation?

Reply:
> is there value in returning invitation?
I think so. We wrote a tool to automate some Github work for our project, and a part of it was inviting some users to a repo and then send the invitation id to the user for them to accept.
For automated work, the information returned from API calls can be valuable, as it saves a subsequent API call.

It's true that it will fail when user is a collaborator. Perhaps it should be a `Maybe RepoInvitation`?